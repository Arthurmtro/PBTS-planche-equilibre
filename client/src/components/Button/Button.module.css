.button {
	background: rgb(var(--button-background));
	font-weight: var(--font-weight);
	color: var(--button-color);
	font-weight: bold;
	padding: 1vw 2vw;
	font-size: 1vw;

	min-width: fit-content;

	border: none;
	border-radius: var(--button-border-radius);

	transition: background-color 0.3s ease-in-out;
	-webkit-transition: background-color 0.3s ease-in-out;
	-moz-transition: background-color 0.3s ease-in-out;
	-o-transition: background-color 0.3s ease-in-out;
}

.button:hover:enabled {
	cursor: pointer;
	box-shadow: 0 0 2vw #c8c9cc00;

	box-shadow: 0 0 1vw rgb(var(--button-background));
}

.button:disabled {
	background-color: rgba(var(--button-background), 0.3);
}
