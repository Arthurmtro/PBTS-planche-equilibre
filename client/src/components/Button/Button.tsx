import { CSSProperties, ReactNode } from "react"
import styles from "./Button.module.css"

interface IButtonParams {
	children?: ReactNode | string
	onClick?: () => void
	color?: "primary" | "secondary" | "white" | "danger"
	disabled?: boolean
	thin?: boolean
}

export default function Button({ children, onClick, color = "primary", disabled = false, thin = false }: IButtonParams) {
	return (
		<button
			disabled={disabled}
			className={styles.button}
			onClick={onClick}
			style={
				{
					"--button-background":
						color === "primary"
							? "var(--color-rgb-primary)"
							: color === "secondary"
								? "var(--color-rgb-secondary)"
								: color === "danger"
									? "223, 13, 13"
									: "255, 255, 255",
					"--button-color": color === "white" ? "#000000" : "#f7f7f7",
					"--font-weight": thin ? "thin" : "bold",
				} as CSSProperties
			}>
			{children}
		</button>
	)
}
