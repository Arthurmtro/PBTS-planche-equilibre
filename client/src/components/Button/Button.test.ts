
import { cleanup, fireEvent, render } from '@testing-library/react';
import Button from './Button';

// Note: running cleanup afterEach is done automatically for you in @testing-library/react@9.0.0 or higher
// unmount and cleanup DOM after the test is finished.
afterEach(cleanup);

function renderButton(props: Partial<any> = {}) {
    const defaultProps: any = {
        onClick() {
            return;
        },
        children: "This is children"
    };
    //return render(<Button{ ...defaultProps } { ...props } />);
}

it('Button changes the text after click', () => {
    const { queryByLabelText, getByLabelText } = render(
        <Button />,
    );

    expect(queryByLabelText(/off/i)).toBeTruthy();

    fireEvent.click(getByLabelText(/off/i));

    expect(queryByLabelText(/on/i)).toBeTruthy();
})