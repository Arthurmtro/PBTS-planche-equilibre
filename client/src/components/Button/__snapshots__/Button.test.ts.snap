// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bouton qui change de fonction et de style 1`] = `
<button
  className="button"
  disabled={false}
  style={
    Object {
      "--button-background": "var(--color-rgb-primary)",
      "--button-color": "#f7f7f7",
      "--font-weight": "bold",
    }
  }
>
  text
</button>
`;

exports[`Bouton qui change de fonction et de style 2`] = `
<button
  className="button"
  disabled={false}
  style={
    Object {
      "--button-background": "var(--color-rgb-primary)",
      "--button-color": "#f7f7f7",
      "--font-weight": "bold",
    }
  }
>
  text
</button>
`;
